src/main/java/com/algorand/algosdk/account/Account.java

```diff
@@ -247,7 +247,7 @@ static public void setFeeByFeePerByte(Transaction tx, BigInteger suggestedFeePer
         if (fee.compareTo(Account.MIN_TX_FEE_UALGOS) < 0) {
             fee = Account.MIN_TX_FEE_UALGOS;
         }
-        tx.setFee(fee);
+        tx.fee = fee;
```

Comment:
nit: a little strange that `tx.fee` is initialized twice in line 243 and here

Reply:
I am unaware of the previous initialization, but this function do not return early, so I am fine with just removing it.

Reply:
ok it is interesting now, seems fee has something to do with `estimateEncodedSize`, so I don't think we should remove it...

Reply:
got it, sounds reasonable!